packages/shared/src/components/search/SearchBar.tsx

```diff
@@ -0,0 +1,327 @@
+            <FieldInput
+              disabled={disabled}
+              placeholder={placeholder}
+              name={name}
+              id={inputId}
+              ref={inputRef}
+              onFocus={(event) => {
+                onFocus();
+                externalOnFocus?.(event);
+              }}
+              onBlur={(event) => {
+                onBlur();
+                externalOnBlur?.(event);
+              }}
+              onInput={onInput}
+              autoFocus={autoFocus}
+              type="primary"
+              aria-describedby={describedBy}
+              autoComplete="off"
+              className={classNames(
+                'flex-1 caret-theme-status-cabbage',
+                getFieldFontColor({ readOnly, disabled, hasInput, focused }),
+              )}
+              required
+            />
+          )}
+
+          <div className="flex gap-3 items-center">
+            {hasInput && (
+              <Button
+                {...rightButtonProps}
+                className="btn-tertiary"
+                buttonSize={ButtonSize.Small}
+                title="Clear query"
+                onClick={onClearClick}
+                icon={<CloseIcon size={IconSize.Small} />}
+                disabled={!hasInput}
+              />
+            )}
+
+            {!isMobile && (
+              <div className="h-8 border border-theme-divider-quaternary" />
+            )}
+
+            <SimpleTooltip
+              content={
+                searchHistory.length === 0
+                  ? 'Your search history is empty'
+                  : 'See search history'
+              }
+            >
+              <div>
+                <Button
+                  {...rightButtonProps}
+                  className="btn-tertiary"
+                  buttonSize={ButtonSize.Small}
+                  title="Search history"
+                  onClick={seeSearchHistory}
+                  icon={<TimerIcon size={IconSize.Small} />}
+                  disabled={searchHistory.length === 0}
+                />
+              </div>
+            </SimpleTooltip>
+            {!isMobile && (
+              <Button
+                {...rightButtonProps}
+                className="btn-primary"
+                buttonSize={ButtonSize.Medium}
+                title="Submit"
+                onClick={onSubmit}
+                icon={<SendAirplaneIcon size={IconSize.Medium} />}
+                disabled={!hasInput}
+              />
+            )}
+          </div>
+        </BaseField>
+
+        <RaisedLabel type={RaisedLabelType.Beta} />
+      </div>
+
+      {!isMobile && showProgress && (
+        <div className="mt-6">
+          <SearchProgressBar progress={progress} />
+
+          {progress > 0 && progress < 100 && (
+            <div className="mt-2 typo-callout text-theme-label-tertiary">
+              🚀 Generating answer
+            </div>
+          )}
+
+          {progress === 100 && completedTime && (
```

Comment:
I think this one alone can tell us it is done.
```suggestion
          {completedTime && (
```